pkg/controller/operators/catalog/operator.go

```diff
@@ -41,10 +41,10 @@ type Operator struct {
 	*queueinformer.Operator
 	client             versioned.Interface
 	namespace          string
-	sources            map[registry.SourceKey]registry.Source
```

Comment:
This is a bigger wrench than I intended to throw when I started looking at this PR...

But it strikes me that this type of tracking (for both "sources" and "subscriptions") is exactly what the cache and indexers are supposed to do for you in client-go. 

We already have SharedIndexInformers for all of this available to the operator, so we can probably just use the index provided there.

Take a look at the cache tooling and see if you agree: https://github.com/kubernetes/client-go/tree/master/tools/cache

I would guess that it's as simple as keeping track of the indexInformers by namespace and then querying those.

Reply:
I think this makes a lot of sense when dealing with subscriptions since we don't need to do any additional work to pull it from the indexer's cache/store. Sources however, need an InMem instance constructed from the catalog source object and its config map before they are useful.

Reply:
Good point!

A future optimization to consider would be to disallow "internal" catalogsources (replacing them with pods that expose an API over the network), which would let us do the same caching for sources and remove the need to parse and keep all the contents in memory.

Reply:
Thanks!

Also, it's questionable at this point whether knowing about subscriptions at resolution time matters as much as knowing what's in the target namespace.

Reply:
Ignore this for this PR, this is something that we should do in general and is orthogonal to this change.

Reply:
Tracked in ALM-679, ignore this for this PR